leakcanary-android-core/src/main/java/leakcanary/DefaultOnHeapAnalyzedListener.kt

```diff
@@ -55,6 +73,50 @@ class DefaultOnHeapAnalyzedListener(private val application: Application) : OnHe
     )
   }
 
+  /**
+   * Android TV devices do not have notifications, therefore the only easy and non-invasive way
+   * to communicate with user is via Toast messages. These are used just to grab user attention and
+   * to direct them to Logcat where a much more detailed report will be printed.
+   */
+  private fun showToast(heapAnalysis: HeapAnalysis) {
+    // Post the Toast into main thread and wrap it with try-catch in case Toast crashes (it happens)
+    handler.post {
+      val toastText: String = when (heapAnalysis) {
+        is HeapAnalysisSuccess -> {
+          val leakTypeCount = heapAnalysis.applicationLeaks.size + heapAnalysis.libraryLeaks.size
+          application.getString(
+              R.string.leak_canary_tv_analysis_success,
+              leakTypeCount,
+              heapAnalysis.libraryLeaks.size
+          )
+        }
+        is HeapAnalysisFailure -> application.getString(R.string.leak_canary_tv_analysis_failure)
+      }
+      Toast.makeText(
+          application,
```

Comment:
See leakcanary.internal.AndroidHeapDumper#showToast

Toasting when no activity is resumed may lead to a crash.

Reply:
I gave it a look, and in the `AndroidHeapDumper` toast is being created with Activity's context. In my case I use the `Application` as a context. As a result, I am able to see Toast appear even when the app is backgrounded, which is ok for TV devices since we want to let users know that memory dump is completed when app is backgrounded.

Reply:
yeaaaah but.. In recent Android versions you can no longer toast when the app is in background, hence the use of activity context and check for resume.

Reply:
Whoa, didn't know about that. It might be a problem because I don't have the reference to the activity here, need to pass it over from somewhere..
Do you have more context on that requirement about backgrounded activities? Looking at `Toast`'s source code in API 29 and it still mentions that passing Application context is ok 😟

Reply:
Passing the app context might be ok but I think you must be visible? And it crashes on some version of Android (that famous toast crash) but not anymore.

Anyway `leakcanary.internal.AndroidHeapDumper#showToast` shows you how to do all that.

Reply:
Oh, so that famous crash might be the reason I had the `try-catch` block in first place. 
Cool, will try to fix it via activity context

Reply:
> even when backgrounded

welp the idea is to only keep the activity reference while the activity is resumed (and therefore won't crash when toasting) so that likely won't work